Dependencies/Onboarding/Sources/Onboarding/Scenes/Intro/IntroViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  IntroViewModel.swift
+//  
+//
+//  Created by MaTooSens on 18/08/2023.
+//
+
+import CoreInterface
+import Foundation
+import OnboardingInterface
+import Utilities
+
+final class IntroViewModel: ViewModelInterface {
+    enum State: Equatable {
+        case idle
+        case loading
+        case error
+        case loaded
+    }
+    
+    enum Event {
+        case getIntros
+        case backButtonTapped
+        case nextButtonTapped
+        case skipButtonTapped
+    }
+    
+    @Published private(set) var state: State = .idle
+    @Published private(set) var currentScreen: Int = 0
+    @Published private(set) var intros: [Intro] = []
+    
+    var isBackButtonVisable: Bool {
+        currentScreen != 0
+    }
+    
+    func trigger(_ event: Event) {
+        switch event {
+        case .getIntros:
+            getIntros()
+        case .backButtonTapped:
+            backButtonTapped()
+        case .nextButtonTapped:
+            nextButtonTapped()
+        case .skipButtonTapped:
+            skipButtonTapped()
+        }
+    }
+}
+
+
+// MARK: Functions
+extension IntroViewModel {
+    private func getIntros() {
+        guard state == .idle else {
+            return
+        }
+        
+        self.state = .loading
+        
+        self.intros = Bundle.module.decode([Intro].self, from: "Intro.json")
+        self.state = .loaded
+    }
+    
+    private func backButtonTapped() {
+        currentScreen -= 1
+    }
+    
+    private func nextButtonTapped() {
+        let isLastScreen = currentScreen == intros.count - 1
+        
+        switch isLastScreen {
+        case true:
+            skipButtonTapped()
+        case false:
+            currentScreen += 1
+        }
```

Comment:
`intros.count - 1` to O(n), obczaj https://developer.apple.com/documentation/swift/array/endindex
to jest O(1), 
```suggestion
        if currentScreen == intros.endIndex - 1 {
            currentScreen += 1
        } else {
            skipButtonTapped()
        }
```